### PR TITLE
[sdk + otlp] CHANGELOG tweaks to cover logging PR feedback

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #4334

## Changes

* Incorporates some CHANGELOG feedback from the previous PR.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -16,8 +16,13 @@
   ([#4375](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4375))
 
 * Tweaked the behavior of the `OpenTelemetryLoggerOptions.ParseStateValues`
-  flag, obsoleted `LogRecord.State` and `LogRecord.StateValues` properties, and
-  added `LogRecord.Body` and `LogRecord.Attributes` properties.
+  flag (attributes are now automatically included for most log messages by
+  default), obsoleted `LogRecord.State` and `LogRecord.StateValues` properties,
+  and added `LogRecord.Body` and `LogRecord.Attributes` properties.
+  ([#4334](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334))
```
Comment (utpilla) [inline src/OpenTelemetry/CHANGELOG.md:22]: nit: We need not add the PR link twice.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Logs/CHANGELOG.md [hunk 2]
```diff
@@ -3,11 +3,17 @@
...
 
 * The `OpenTelemetryLoggerOptions.AddOtlpExporter` extension no longer
-  automatically set `OpenTelemetryLoggerOptions.ParseStateValues` to `true`.
+  automatically sets `OpenTelemetryLoggerOptions.ParseStateValues` to `true`.
+  The OpenTelemetry SDK now automatically sets `Attributes` (aka `StateValues`)
+  for the common cases where `ParseStateValues` was previously required.
+  `ParseStateValues` can be set to `true` manually by users to enable parsing of
+  custom states which do not implement `IReadOnlyList` / `IEnumerable`
+  interfaces.
   ([#4334](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334))
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Logs/CHANGELOG.md:12]: Same as https://github.com/open-telemetry/opentelemetry-dotnet/pull/4415/files#r1173147109

File: src/OpenTelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -16,8 +16,13 @@
   ([#4375](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4375))
 
 * Tweaked the behavior of the `OpenTelemetryLoggerOptions.ParseStateValues`
-  flag, obsoleted `LogRecord.State` and `LogRecord.StateValues` properties, and
-  added `LogRecord.Body` and `LogRecord.Attributes` properties.
+  flag (attributes are now automatically included for most log messages by
```
Comment (utpilla) [inline src/OpenTelemetry/CHANGELOG.md:19]: There is also the change that from now on `State` could be `null` even when `ParseStateValues` is set to `false`. (when state is `IReadOnlyList<KeyValuePair<string, object?>>` or `IEnumerable<KeyValuePair<string, object?>>`)

I think we should mention this change as well.

File: src/OpenTelemetry/CHANGELOG.md [hunk 4]
```diff
@@ -16,8 +16,13 @@
...
+  default), obsoleted `LogRecord.State` and `LogRecord.StateValues` properties,
+  and added `LogRecord.Body` and `LogRecord.Attributes` properties.
+  ([#4334](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334))
+
+* If a template (`{OriginalFormat}` attribute) cannot be found on log messages a
```
Comment (utpilla) [inline src/OpenTelemetry/CHANGELOG.md:24]: We could make this more clear:

Tweaked the behavior of the `OpenTelemetryLoggerOptions.IncludeFormattedMessage`: `LogRecord.FormattedMessage` will now automatically be generated, if a template (`{OriginalFormat}` attribute) is not found on log messages (regardless of whether `OpenTelemetryLoggerOptions.IncludeFormattedMessage` is set to `true` or `false`).

File: src/OpenTelemetry/CHANGELOG.md (hunk 3 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/CHANGELOG.md]: I just pushed a new version. LMK what you think.

File: src/OpenTelemetry/CHANGELOG.md (hunk 4 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/CHANGELOG.md:41]: I just pushed a new version. LMK what you think.

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/CHANGELOG.md:23]: I kind of did this on purpose. I think it reads more cleanly with conceptual changes broken out into their own simple(ish) bullets/lines. I could roll everything into one bullet, but having duplicate PR links doesn't seem like a big deal to me. LMK if you still want a change.

Comment (utpilla) [reply src/OpenTelemetry/CHANGELOG.md:23]: I don't have a strong opinion here. I was suggesting we use multiple bullet points but add the PR link only once at the end of the last bullet point. Someone might complain about unclaimed bullet points then if they don't read the CHANGELOG the way we expect them to.

**Review by utpilla — APPROVED** [review]

**Pull request merged**